src/main/java/org/checkerframework/specimin/unsolved/UnsolvedSymbolGenerator.java

```diff
@@ -988,6 +999,121 @@ private void handleConstructorCall(
+            JavaParserUtil.getTypeFromQualifiedName(
+                resolvedType.describe(), fqnsToCompilationUnits);
+
+        if (parentClass == null) {
+          return;
+        }
+
+        potentialScopeFQNs =
+            fullyQualifiedNameGenerator
+                .getFQNsOfAllUnresolvableParents(parentClass, methodDecl)
+                .values();
+      } catch (UnsolvedSymbolException ex) {
+        potentialScopeFQNs =
+            Set.of(fullyQualifiedNameGenerator.getFQNsFromType(anonClass.getType()).erasedFqns());
+      }
+    } else {
+      potentialScopeFQNs =
+          fullyQualifiedNameGenerator
+              .getFQNsOfAllUnresolvableParents(
+                  JavaParserUtil.getEnclosingClassLike(methodDecl), methodDecl)
+              .values();
+    }
+
+    if (potentialScopeFQNs.isEmpty()) {
+      // If there are no potential scope FQNs, then this method is likely an override of a method
+      // in an existing class or JDK interface
+      return;
+    }
+
+    String simpleSignature = methodDecl.getNameAsString() + "(";
+
+    for (Parameter param : methodDecl.getParameters()) {
+      simpleSignature +=
+          JavaParserUtil.getSimpleNameFromQualifiedName(
+              JavaParserUtil.erase(param.getTypeAsString()));
+    }
+
+    simpleSignature += ")";
+
+    Set<String> potentialMethodFQNs = new LinkedHashSet<>();
+    for (Set<String> set : potentialScopeFQNs) {
+      for (String fqn : set) {
+        potentialMethodFQNs.add(fqn + "#" + simpleSignature);
+      }
+    }
+
+    UnsolvedMethodAlternates generated =
+        (UnsolvedMethodAlternates) findExistingAndUpdateFQNs(potentialMethodFQNs);
+
+    if (generated != null) {
+      return;
+    }
+
+    List<MemberType> parameters = new ArrayList<>();
+    for (Parameter param : methodDecl.getParameters()) {
+      MemberType paramType =
+          getOrCreateMemberTypeFromFQNs(
+              fullyQualifiedNameGenerator.getFQNsFromType(param.getType()));
+      parameters.add(paramType);
+    }
+
+    List<UnsolvedClassOrInterfaceAlternates> potentialDeclaringTypes = new ArrayList<>();
+
+    for (Set<String> fqns : potentialScopeFQNs) {
+      potentialDeclaringTypes.add(findExistingAndUpdateFQNsOrCreateNewType(fqns));
+    }
+
+    MemberType returnType =
+        getOrCreateMemberTypeFromFQNs(
+            fullyQualifiedNameGenerator.getFQNsFromType(methodDecl.getType()));
+
+    List<MemberType> exceptions = new ArrayList<>();
+    for (ReferenceType exception : methodDecl.getThrownExceptions()) {
+      MemberType exceptionType =
+          getOrCreateMemberTypeFromFQNs(fullyQualifiedNameGenerator.getFQNsFromType(exception));
+      exceptions.add(exceptionType);
+    }
+
+    AccessSpecifier specifier = methodDecl.getAccessSpecifier();
+    String accessModifier =
+        switch (specifier) {
+          case PUBLIC -> "public";
+          case PROTECTED -> "protected";
+          case PRIVATE -> "private";
```

Comment:
The `PRIVATE` case isn't possible, right, since this method must be an override? I would throw here rather than proceeding if we do hit this case, because I think it's an error condition.